### PR TITLE
adi - User and Application Schema Defined

### PR DESCRIPTION
**Pull request by adityaavadhanam2002**
created fields for things needed to be kept track of for both Users and Applications.

### CODE REVIEW

**Comment by Jsbbvk** [comment]
Make sure you pull latest from `main`

File: src/models/Application.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { Schema, model, models } from 'mongoose'
+import type { User } from 'next-auth'
+
+enum Decision {
+  'Accepted',
+  'Not Accepted',
```
Comment (softwaresale) [inline src/models/Application.ts]: You might consider calling this 'rejected' or 'denied' or something. I think that one word keywords are easier to check and deal with

File: src/models/User.ts [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
 import type { User } from 'next-auth'
 
+enum Role {
+  'Applicant',
+  'Hacker',
+  'Exec',
+  'Sponsor',
+}
+
 export interface IUser extends User {
```
Comment (softwaresale) [inline src/models/User.ts:12]: do we need a field for "user has filled out profile info"?

**Review by softwaresale — COMMENTED** [review]
Don't forget to either rebase on main (`git rebase origin/main`) or merge main into your branch

File: src/models/User.ts (hunk 2 above)
Comment (skaur728) [reply src/models/User.ts:12]: yes we would need something like that to track where they're at

**Review by softwaresale — APPROVED** [review]

**Pull request merged**